### PR TITLE
fix onBlur in ComboBox when no Button provided

### LINKED ISSUE #6007: ComboBox: onBlur doesn't work when no Button provided
### Provide a general summary of the issue here

`onBlur` is not triggered for blur event, when `buttonRef` is empty in `useComboBox` hook

### 🤔 Expected Behavior?

`onBlur` should be triggered, even when there is no button inside. 

### 😯 Current Behavior

You must use `Button` inside of `ComboBox` to onBlur work correctly. 

### 💁 Possible Solution

Check if the `buttonRef` is filled
https://github.com/adobe/react-spectrum/blob/361d47080c79e4d52ee287078b9e3c27ed312f2e/packages/%40react-aria/combobox/src/useComboBox.ts#L161

### 🔦 Context

_No response_

### 🖥️ Steps to Reproduce

https://codesandbox.io/p/sandbox/exciting-river-rfnxhh

### Version

1.1.1

### What browsers are you seeing the problem on?

Chrome

### If other, please specify.

_No response_

### What operating system are you using?

Mac

### 🧢 Your Company/Team

_No response_

### 🕷 Tracking Issue

_No response_

### PR DESCRIPTION
**Pull request by tomekancu**
Closes #6007

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

### CODE REVIEW

**Comment by snowystinger** [comment]
GET_BUILD

File: packages/@react-aria/combobox/src/useComboBox.ts [hunk 1]
```diff
@@ -157,8 +157,8 @@ export function useComboBox<T>(props: AriaComboBoxOptions<T>, state: ComboBoxSta
   };
 
   let onBlur = (e: FocusEvent<HTMLInputElement>) => {
-    // Ignore blur if focused moved to the button or into the popover.
-    if (e.relatedTarget === buttonRef?.current || popoverRef.current?.contains(e.relatedTarget)) {
+    // Ignore blur if focused moved to the button(if exists) or into the popover.
+    if ((buttonRef?.current && e.relatedTarget === buttonRef?.current) || popoverRef.current?.contains(e.relatedTarget)) {
```
Comment (ryo-manba) [inline packages/@react-aria/combobox/src/useComboBox.ts:161]:
```suggestion
    let blurFromButton = buttonRef.current && buttonRef.current === e.relatedTarget;
    let blurIntoPopover = popoverRef.current?.contains(e.relatedTarget);
    // Ignore blur if focused moved to the button(if exists) or into the popover.
    if (blurFromButton || blurIntoPopover) {
      return;
    }
```

Splitting the logic into separate checks could make things a bit clearer to read!

File: packages/@react-aria/combobox/test/useComboBox.test.js [hunk 2]
```diff
@@ -132,6 +132,18 @@ describe('useComboBox', function () {
     expect(toggleSpy).toHaveBeenLastCalledWith(null, 'manual');
   });
 
+  it('should not prevent onBlur when no button provided', function () {
```
Comment (snowystinger) [inline packages/@react-aria/combobox/test/useComboBox.test.js:135]: generally like to name it for what it does, not for the bug it was fixing
```suggestion
  it('should call onBlur when no button provided and you leave the field', function () {
```

**Review by snowystinger — APPROVED** [review]

File: packages/@react-aria/combobox/test/useComboBox.test.js (hunk 2 above)
Comment (tomekancu) [reply packages/@react-aria/combobox/test/useComboBox.test.js]: good to know, thank you ;)

**Review by LFDanLu — APPROVED** [review]
LGTM, thanks for the fix!

**Pull request merged**